dotCMS/src/main/java/com/dotcms/content/elasticsearch/ESQueryCache.java

```diff
@@ -0,0 +1,110 @@
+package com.dotcms.content.elasticsearch;
+
+import java.util.Optional;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.search.SearchHits;
+import com.dotcms.enterprise.license.LicenseManager;
+import com.dotmarketing.business.Cachable;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.util.Config;
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * This cache will take an ElasticSearch SearchRequest as a key and will return the SearchHits for
+ * that SearchRequest from cache if they are available.  This entire cache can be turned off by setting
+ * ES_CACHE_SEARCH_QUERIES = false in your dotmarketing-config.properties
+ * 
+ * @author will
+ *
+ */
+public class ESQueryCache implements Cachable {
+
+
+    private final DotCacheAdministrator cache;
+
+    private final boolean shouldUseCache;
+
+    @VisibleForTesting
+    public ESQueryCache(DotCacheAdministrator cache, boolean shouldUseCache) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'shouldUseCache' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5288908)